node/resolvers/saveIndex.ts

```diff
@@ -1,18 +1,41 @@
 import { EXTENDED_INDEX_FILE, getBucket, hashString } from '../utils'
-import { currentDate, SitemapIndex } from './../middlewares/generateMiddlewares/utils'
+import {
+  currentDate,
+  SitemapIndex,
+} from './../middlewares/generateMiddlewares/utils'
 import { getDefaultStoreBinding } from './../resources/bindings'
 
-export const saveIndex = async (_: {}, { index, binding }: { index: string, binding?: string }, ctx: Context) => {
-  const { clients: { vbase } } = ctx
+export const saveIndex = async (
+  _: {},
+  { index, binding }: { index: string; binding?: string },
+  ctx: Context
+) => {
+  const {
+    clients: { vbase },
+  } = ctx
+  let success = false
   if (!binding) {
     binding = await getDefaultStoreBinding(ctx)
   }
   const bucket = getBucket('', hashString(binding))
-  const { index: extendedIndexes } = await vbase.getJSON<SitemapIndex>(bucket, EXTENDED_INDEX_FILE, true) ||  { index: [] }
-  extendedIndexes.push(index)
-  await vbase.saveJSON(bucket, EXTENDED_INDEX_FILE, {
-    index: extendedIndexes,
-    lastUpdated: currentDate(),
-  })
-  return true
+  const { index: extendedIndexes } = (await vbase.getJSON<SitemapIndex>(
+    bucket,
+    EXTENDED_INDEX_FILE,
+    true
+  )) || { index: [] }
+  // check if index is already in the extendedIndexes
+  if (!extendedIndexes.includes(index)) {
+    extendedIndexes.push(index)
+  }
+  await vbase
```

Comment:
If it is already on the extendedIndexes, should we be running this `saveJSON` anyway?

Reply:
I think it's still worth running the `saveJSON` in order to eliminate any duplicate indexes that may have been added previously. Does that make sense?

Reply:
Oh, now I understand what you did there. 
Wouldn't the duplicates be eliminated in the next sitemap generation (the changes you made on `sitemap.ts`)?

Reply:
Well, in `sitemap.ts` the indexes are loaded from vbase, then any duplicates are removed, then the sitemap is generated. But the code doesn't save the de-duped indexes back to vbase, so they still exist.